test/idp/src/main/kotlin/org/codice/compliance/web/sso/RedirectLoginTest.kt

```diff
@@ -76,8 +78,11 @@ class RedirectLoginTest : StringSpec() {
                     .`when`()
                     .get(Common.getSingleSignOnLocation(SamlProtocol.REDIRECT_BINDING))
 
+            // Get response from plugin portion
             val idpResponse = getServiceProvider(IdpResponder::class.java).getIdpRedirectResponse(response)
-            verifyResponse(idpResponse, true)
+            idpResponse.isRelayStateGiven = true
```

Comment:
❗️ You are not setting this in the `PostLoginTest`.
❓  This is nitpicky, but I want this `isRelayStateGiven` property to be `protected` according to my other comments. So my initial thought is to put the set operation in the `IdpResponse` builders and we would pass in the boolean into the `getBINDINGResponse` here in the test. Because only the test knows what the value should be. But then we're relying on the plugin implementers to pass that boolean into the builder which isn't great... We could initialize the Builder object for them and pass it in, but that would be weird IMO. I'm open to ideas, or we may just have to make an exception for making `isRelayStateGiven` `protected`. Thoughts?

Reply:
Ooo, I can make `isRelayStateGiven` private. I'll do that.

Back to the topic at hand, I chose this way because:
1. I hate flags. I use them so often but I hate them.
2. I believe the `isRelayStateGiven` is information only we know, therefore it should be our responsibility and our's alone to set it
3. Builders _usually_ initialize the builder and build from scratch, therefore I thought passing in a builder for them to use would be kind of odd.

But of course this is open for discussion @blen-desta @coyotesqrl .